src/app/components/shell/header/header.js

```diff
@@ -1,21 +1,88 @@
+import $ from '~/helpers/$';
 import BaseView from '~/helpers/backbone/view';
 import {on, props} from '~/helpers/backbone/decorators';
 import {template} from './header.hbs';
 
-function toggleClass(el, name) {
-    if (el.classList.contains(name)) {
-        el.classList.remove(name);
-    } else {
-        el.classList.add(name);
-    }
-}
-
 @props({
     el: '#header',
     template: template
 })
 export default class Header extends BaseView {
 
+    constructor() {
+        super(...arguments);
+
+        this.meta = {};
+
+        this.templateHelpers = {
+            collapsed: () => this.meta.collapsed,
+            fixed: () => this.meta.fixed,
+            transparent: () => this.meta.transparent
+        };
+    }
+
+    classList(action, ...args) {
+        var result = null;
+        let header = this.el.querySelector('.page-header');
+
+        if (header && typeof header.classList === 'object') {
+            result = header.classList[action](...args);
+        }
+
+        return result;
+    }
+
+    collapse() {
+        this.meta.collapsed = true;
+        this.classList('add', 'collapsed');
+        return this;
+    }
+
+    pin() {
+        this.meta.fixed = true;
+        this.classList('add', 'fixed');
+        return this;
+    }
+
+    transparent() {
+        this.meta.transparent = true;
+        this.classList('add', 'transparent');
+        return this;
+    }
+
+    reset() {
+        this.meta.collapsed = false;
+        this.meta.pinned = false;
+        this.meta.transparent = false;
+        this.classList('remove', 'collapsed');
+        this.classList('remove', 'fixed');
+        this.classList('remove', 'transparent');
+        return this;
+    }
+
+    isCollapsed() {
+        return !!this.classList('contains', 'collapsed');
+    }
+
+    isPinned() {
+        return !!this.classList('contains', 'fixed');
+    }
+
+    isTransparent() {
+        return !!this.classList('contains', 'transparent');
+    }
+
+    get height() {
+        let header = this.el.querySelector('.page-header');
```

Comment:
Q: I've been doing one `let` with a comma-separated list of declarations; you use one `let` per declaration. Do we want to standardize, or does it not matter?